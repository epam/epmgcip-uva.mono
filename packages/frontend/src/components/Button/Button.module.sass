@use 'src/style/colors'

.button 
  background-color: inherit
  border: none
  padding: 0
  display: flex
  justify-content: flex-end

  &:hover
    cursor: pointer

  &:active
    opacity: 0.85

  &:disabled,
  &[disabled]
    background-color: colors.$gray-light
