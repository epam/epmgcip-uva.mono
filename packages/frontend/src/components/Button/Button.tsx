import { getClassesList } from 'src/utils/getClassesList';
import css from './Button.module.sass';

interface ButtonProps {
  children: JSX.Element | string;
  onClick: () => void;
  className?: string;
  id?: string;
  disabled?: boolean;
}

export const Button = ({
  children,
  onClick,
  className,
  id,
  disabled,
}: ButtonProps) => {
  const buttonClasses = getClassesList(css.button, className);

  return (
    <button
      onClick={onClick}
      className={buttonClasses}
      id={id}
      disabled={disabled}
    >
      {children}
    </button>
  );
};
