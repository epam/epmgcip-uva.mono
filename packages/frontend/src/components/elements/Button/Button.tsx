import { getClassesList } from 'src/utils/getClassesList';
import css from './Button.module.sass';

interface ButtonProps {
  children: JSX.Element | string | number;
  onClick?: () => void;
  className?: string;
  id?: string;
  disabled?: boolean;
  stopPropagation?: boolean;
}

export const Button = ({
  children,
  onClick,
  className,
  id,
  disabled,
  stopPropagation,
}: ButtonProps) => {
  const buttonClasses = getClassesList(css.button, className);

  return (
    <button
      onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
        if (stopPropagation) {
          e.stopPropagation();
          e.preventDefault();
        }
        onClick && onClick();
      }}
      className={buttonClasses}
      id={id}
      disabled={disabled}
    >
      {children}
    </button>
  );
};
