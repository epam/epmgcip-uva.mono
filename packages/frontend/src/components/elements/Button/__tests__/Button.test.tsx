import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from '../Button';

const textMock = 'Test Button';
const childrenMock = <div>{textMock}</div>;
const onClickMock = vi.fn();
const additionalClassMock = 'additionalTestClass';

describe('Testing: Button', () => {
  it('should render', () => {
    const { container } = render(
      <Button children={childrenMock} onClick={onClickMock} className={additionalClassMock} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should call provided function after click', () => {
    render(
      <Button children={childrenMock} onClick={onClickMock} />
    );

    const renderedButton = screen.getByText(textMock);

    fireEvent.click(renderedButton);

    expect(onClickMock).toHaveBeenCalled();
  });
});
