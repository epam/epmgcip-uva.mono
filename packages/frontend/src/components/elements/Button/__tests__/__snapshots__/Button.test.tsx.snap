// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: Button > should render 1`] = `
<div>
  <button
    class="_button_2842f2 additionalTestClass"
  >
    <div>
      Test Button
    </div>
  </button>
</div>
`;
